tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -0,0 +1,303 @@
+
+                    ranges = Ranges.MergeRanges(length, ranges, parameterTainted.Ranges);
+                }
+
+                length += currentParameter!.Length;
+            }
+
+            if (ranges != null)
+            {
+                taintedObjects.Taint(result, ranges);
+            }
+        }
+        catch (Exception err)
+        {
+            Log.Error(err, "StringModuleImpl.OnstringConcat(StringConcatParams) exception {Exception}", err.Message);
+        }
+
+        return result;
+    }
+
+    /// <summary> Overload for multi params (up to 5) </summary>
+    /// <param name="parameters"> StringConcat params struct </param>
+    /// <param name="result"> Result </param>
+    /// <returns> result </returns>
+    public static string OnStringConcat(IEnumerable parameters, string result)
+    {
+        try
+        {
+            if (!CanBeTainted(result))
+            {
+                return result;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext == null)
+            {
+                return result;
+            }
+
+            TaintedObjects taintedObjects = iastContext.GetTaintedObjects();
+
+            Range[]? ranges = null;
+            int length = 0;
+            foreach (var parameterAsObject in parameters)
+            {
+                var currentParameter = parameterAsObject?.ToString();
+                if (!CanBeTainted(currentParameter))
+                {
+                    continue;
+                }
+
+                var taintedParameter = GetTainted(taintedObjects, currentParameter);
+                if (taintedParameter != null)
+                {
+                    if (ranges == null)
+                    {
+                        if (taintedParameter != null)
+                        {
+                            if (length == 0)
+                            {
+                                ranges = taintedParameter.Ranges;
+                            }
+                            else
+                            {
+                                ranges = new Range[taintedParameter!.Ranges!.Length];
+                                Ranges.CopyShift(taintedParameter!.Ranges, ranges, 0, length);
+                            }
+                        }
+
+                        length += currentParameter!.Length;
+                        continue;
+                    }
+
+                    ranges = Ranges.MergeRanges(length, ranges, taintedParameter.Ranges);
+                }
+
+                length += currentParameter!.Length;
+            }
+
+            if (ranges != null)
+            {
+                taintedObjects.Taint(result, ranges);
+            }
+        }
+        catch (Exception err)
+        {
+            Log.Error(err, "StringModuleImpl.OnstringConcat(IEnumerable) exception {Exception}", err.Message);
+        }
+
+        return result;
+    }
+
+    public readonly struct StringConcatParams
```

Comment:
I think this fulfils the criteria to be a `ref struct`

Reply:
Yes, that's right. Done!